kratos/sources/constitutive_law.cpp

```diff
@@ -57,6 +57,8 @@ namespace Kratos
     KRATOS_CREATE_LOCAL_FLAG( ConstitutiveLaw, ISOTROPIC,                   8 );
     KRATOS_CREATE_LOCAL_FLAG( ConstitutiveLaw, ANISOTROPIC,                 9 );
 
+	KRATOS_CREATE_LOCAL_FLAG(ConstitutiveLaw, STENBERG_STABILIZATION_SUITABLE,	10);

```

Comment:
@RiccardoRossi this is the flag that we discussed already. Please have a look at the file diffs, then it is clear what the intended usage is

Reply:
Hi Philip,

honestly i am quite against adding the flag STENBERG_STABILIZATION_SUITABLE.

To my (little) understanding is something very specific to this case, and not something general to many constitutive laws.
My proposal would be to pass the StenbergStabilization through the properties, the same way you do for say the YOUNG_MODULUS.

I appreciate this would not do if you want the element to provide to the ConstitutiveLaw an information of wheter it wants or not this stabilization to be used. Is that the case?

Reply:
I understand
The problem is that this Stabilization might not work right away with fancier Claws. We contacted Massimo abt including this Stabilization and he said this.
It works perfect for the tested laws, but the user should at least be aware that he has to check his results carefully if another (not yet implemented) Claw is used.
Can we use `Kratos::Components` somehow?

Reply:
i don't have an answer for this. What would you need the kratos components for?

Reply:
I have a proposal:

you could create two different CLs: 

like StenbergStabilizedNewLaw and NormalLaw

then the StenbergStabilizedNewLaw would implement a function 

       Calculate(STENBERG_STABILIZATION_SUITABLE)

where STENBERG_STABILIZATION_SUITABLE would be a Variable<bool>

which the element would call to ask if the stabilization is there or not.

if this is only to be used in a check it shouldn't matter, and you are not introducing a special flag for that. What do you think?

Reply:
Thx for the suggestion.
I have an even better suggestion (bcs I don't like to copy a CLaw just bcs of the Stabilization):
As you suggested, STENBERG_STABILIZATION_SUITABLE is a variable <Bool>
The CLaws for which the Stabilization has been checked get a  CalculateValue Function:

```
bool& ConstitutiveLaw::CalculateValue(Parameters& rParameterValues, const Variable<bool>& rThisVariable, bool& rValue)
{
    if (rThisVariable== STERNBERG_STABILIZATION_SUITABLE)
        return true;
    else 
        return rValue
}
```
This way no other claws are affected

for this to work we have to add the `CalculateValue` function with template argument `bool` to the base class.

What do you think?

Reply:
ping @RiccardoRossi

Reply:
ok, this sounds fine to me